### PR TITLE
Return structured success/error json responses

### LINKED ISSUE #7: Add infrastructure for json response

### PR DESCRIPTION
**Pull request by tokland**
Closes #7 

(depends on #9)

### CODE REVIEW

**Review by adrianq — APPROVED** [review]
@tokland Looks great as usual! :) Just a single comment, what do you think about adding in the doc a few examples with different responses and a very brief description of the response structure?

@ifoche you said the Angular call you finally used was slightly different, right? Do you maybe wanna add a commit to this PR and correct the docu?

File: test_fire.py [hunk 1]
```diff
@@ -40,20 +40,40 @@ def request(self, method, path, data=None, user=None):
...
+    ### JSON structure
+
+    def test_existing_resource(self):
+        res = self.request("GET", '/users/3', user=self.USERS["marilyn"])
+        self.assertEqual(res.status, 200)
+        self.assertEqual(res.body.get("status"), "success")
+        self.assertTrue(res.body.get("data"))
+
+    def test_not_found(self):
+        res = self.request("GET", '/thisEndPointThoseNotExist')
```
Comment (ifoche) [inline test_fire.py]: I guess it's a typo and should read '/thisEndPointDoesNotExist' right?

**Review by ifoche — CHANGES_REQUESTED** [review]
@tokland it's a minimum change what I'm asking for...only a typo. Thank you very much!

**Review by ifoche — APPROVED** [review]
Thanks @tokland

**Pull request merged**